File: groclist-api-dotnet/GrocListApi.Core/ApiModels/ChangePasswordModel.cs
using System.ComponentModel.DataAnnotations;

namespace GrocListApi.Core.ApiModels
{
    public class ChangePasswordModel
    {
        public required string Password { get; set; }
        public required string NewPassword { get; set; }
        [Display(Name = "Confirm Password")]
        [Compare("NewPassword", ErrorMessage = "Passwords do not match.")]
        public required string ConfirmNewPassword { get; set; }
    }
}

File: groclist/src/app/models/change-password-model.ts
export class ChangePasswordModel {
    public password: string;
    public newPassword: string;
    public confirmNewPassword: string;

    constructor(password: string, newPassword: string, confirmNewPassword: string) {
        this.password = password;
        this.newPassword = newPassword;
        this.confirmNewPassword = confirmNewPassword;
    }
}
